webapp/components/emoji_picker/emoji_picker.jsx

```diff
@@ -0,0 +1,410 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+
+import $ from 'jquery';
+import * as Emoji from 'utils/emoji.jsx';
+import EmojiStore from 'stores/emoji_store.jsx';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import ReactDOM from 'react-dom';
+import * as Utils from 'utils/utils.jsx';
+import ReactOutsideEvent from 'react-outside-event';
+import {FormattedMessage} from 'react-intl';
+
+import EmojiPickerCategory from './components/emoji_picker_category.jsx';
+import EmojiPickerItem from './components/emoji_picker_item.jsx';
+import EmojiPickerPreview from './components/emoji_picker_preview.jsx';
+
+// This should include all the categories available in Emoji.CategoryNames
+const CATEGORIES = [
+    'recent',
+    'people',
+    'nature',
+    'food',
+    'activity',
+    'travel',
+    'objects',
+    'symbols',
+    'flags',
+    'custom'
+];
+
+class EmojiPicker extends React.Component {
+    static propTypes = {
+        customEmojis: React.PropTypes.object,
+        onEmojiClick: React.PropTypes.func.isRequired,
+        topOrBottom: React.PropTypes.string.isRequired,
+        emojiOffset: React.PropTypes.number,
+        outsideClick: React.PropTypes.func
+    }
+
+    constructor(props) {
+        super(props);
+
+        // All props are primitives or treated as immutable
+        this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
+
+        this.handleCategoryClick = this.handleCategoryClick.bind(this);
+        this.handleFilterChange = this.handleFilterChange.bind(this);
+        this.handleItemOver = this.handleItemOver.bind(this);
+        this.handleItemOut = this.handleItemOut.bind(this);
+        this.handleItemClick = this.handleItemClick.bind(this);
+        this.handleScroll = this.handleScroll.bind(this);
+        this.handleItemUnmount = this.handleItemUnmount.bind(this);
+        this.renderCategory = this.renderCategory.bind(this);
+        this.onOutsideEvent = this.onOutsideEvent.bind(this);
+
+        this.state = {
+            category: 'recent',
+            filter: '',
+            selected: null
+        };
+    }
+
+    componentDidMount() {
+        this.searchInput.focus();
+    }
+
+    onOutsideEvent = (event) => {
+        // Handle the event.
+        this.props.outsideClick(event);
+    }
+
+    handleCategoryClick(category) {
+        const items = ReactDOM.findDOMNode(this.refs.items);
```

Comment:
I don't mean remove jquery (although I would like that but we've already discussed that) just replace `ReactDOM.findDOMNode(this.refs.items)` with `this.refs.items` because it's the same thing now. [If there's a ref to it, there's no need to use findDOMNode](https://facebook.github.io/react/docs/react-dom.html#finddomnode)